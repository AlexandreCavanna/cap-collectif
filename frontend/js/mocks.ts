/* eslint-env jest */
import type { IntlShape, MessageDescriptor } from 'react-intl'
import 'react-intl'
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay'

export const $refType: any = null
export const $fragmentRefs: any = null
export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: (message: MessageDescriptor) => String(message),
  formatPlural: (message: string) => String(message),
  formatNumber: (message: string) => String(message),
  formatRelative: (message: string) => String(message),
  formatTime: (message: string) => String(message),
  formatDate: (message: string) => String(message),
  formatMessage: (message: MessageDescriptor) => String(message.id),
}
export const formMock: ReduxFormFormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  // initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  updateSyncErrors: jest.fn(),
  valid: true,
  warning: null,
}
const environment = {
  UNSTABLE_getDefaultRenderPolicy: jest.fn(),
  __log: jest.fn(),
  getOperationTracker: jest.fn(),
  isRequestActive: jest.fn(),
  isServer: jest.fn(),
  options: jest.fn(),
  applyMutation: jest.fn(),
  sendMutation: jest.fn(),
  lookup: jest.fn(),
  sendQuery: jest.fn(),
  subscribe: jest.fn(),
  streamQuery: jest.fn(),
  retain: jest.fn(),
  unstable_internal: {
    areEqualSelectors: jest.fn(),
    createFragmentSpecResolver: jest.fn(),
    createOperationDescriptor: jest.fn(),
    getDataIDsFromFragment: jest.fn(),
    getDataIDsFromObject: jest.fn(),
    getFragment: jest.fn(),
    getPluralSelector: jest.fn(),
    getRequest: jest.fn(),
    getSelector: jest.fn(),
    getSelectorsFromObject: jest.fn(),
    getSingularSelector: jest.fn(),
    getVariablesFromFragment: jest.fn(),
    getVariablesFromObject: jest.fn(),
    getVariablesFromPluralFragment: jest.fn(),
    getVariablesFromSingularFragment: jest.fn(),
    isFragment: jest.fn(),
    isRequest: jest.fn(),
  },
  applyUpdate: jest.fn(),
  check: jest.fn(),
  commitPayload: jest.fn(),
  commitUpdate: jest.fn(),
  execute: jest.fn(),
  executeMutation: jest.fn(),
  executeWithSource: jest.fn(),
  getNetwork: jest.fn(),
  getStore: jest.fn(),
  areEqualSelectors: jest.fn(),
  createFragmentSpecResolver: jest.fn(),
  createOperationDescriptor: jest.fn(),
  getDataIDsFromFragment: jest.fn(),
  getDataIDsFromObject: jest.fn(),
  getFragment: jest.fn(),
  getPluralSelector: jest.fn(),
  getRequest: jest.fn(),
  getSelector: jest.fn(),
  getSelectorsFromObject: jest.fn(),
  getSingularSelector: jest.fn(),
  getVariablesFromFragment: jest.fn(),
  getVariablesFromObject: jest.fn(),
  getVariablesFromPluralFragment: jest.fn(),
  requiredFieldLogger: jest.fn(),
  revertUpdate: jest.fn(),
  replaceUpdate: jest.fn(),
}
export const relayPaginationMock: RelayPaginationProp = {
  environment,
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
}
export const relayRefetchMock: RelayRefetchProp = {
  environment,
  refetch: jest.fn(),
}
export const googleAddressMock = {
  lat: 43.6424564,
  lng: -79.3755156,
  json: '[{"formatted_address":"10 Yonge St, Toronto, ON M5E 1R4, Canada","geometry":{"location":{"lat":43.6424564,"lng":-79.3755156},"location_type":"ROOFTOP","viewport":{"south":43.6411074197085,"west":-79.37686458029151,"north":43.6438053802915,"east":-79.37416661970849}},"types":["street_address"],"address_components":[{"long_name":"10","short_name":"10","types":["street_number"]},{"long_name":"Yonge Street","short_name":"Yonge St","types":["route"]},{"long_name":"Old Toronto","short_name":"Old Toronto","types":["political","sublocality","sublocality_level_1"]},{"long_name":"Toronto","short_name":"Toronto","types":["locality","political"]},{"long_name":"Toronto Division","short_name":"Toronto Division","types":["administrative_area_level_2","political"]},{"long_name":"Ontario","short_name":"ON","types":["administrative_area_level_1","political"]},{"long_name":"Canada","short_name":"CA","types":["country","political"]},{"long_name":"M5E 1R4","short_name":"M5E 1R4","types":["postal_code"]}],"place_id":"ChIJrbgtxSvL1IkR2xGe7JZyuw0","plus_code":{"compound_code":"JJRF+XQ Toronto, ON, Canada","global_code":"87M2JJRF+XQ"}}]',
}
