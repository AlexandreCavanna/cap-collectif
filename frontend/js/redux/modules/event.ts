import type { Exact, Action } from '../../types'
export type State = {
  readonly eventSelected: string | null | undefined
  readonly isMobileListView: boolean
  readonly showEventCreateModal: boolean
}
const initialState: State = {
  eventSelected: null,
  isMobileListView: true,
  showEventCreateModal: false,
}
type ChangeEventSelectedAction = {
  type: 'event/CHANGE_EVENT_SELECTED'
  eventSelected: string | null | undefined
}
type ChangeEventMobileListView = {
  type: 'event/CHANGE_EVENT_MOBILE_VIEW'
  isMobileListView: boolean
}
export type EventAction =
  | ChangeEventSelectedAction
  | ChangeEventMobileListView
  | {
      type: 'event/CHANGE_FILTER'
      filter: string
      value: string
    }
export const changeEventSelected = (eventSelected: string | null | undefined): ChangeEventSelectedAction => ({
  type: 'event/CHANGE_EVENT_SELECTED',
  eventSelected,
})
export const changeEventMobileListView = (isMobileListView: boolean): ChangeEventMobileListView => ({
  type: 'event/CHANGE_EVENT_MOBILE_VIEW',
  isMobileListView,
})
export const reducer = (state: State = initialState, action: Action): Exact<State> => {
  switch (action.type) {
    case '@@INIT':
      return { ...initialState, ...state }

    case 'event/CHANGE_EVENT_SELECTED':
      return { ...state, eventSelected: action.eventSelected }

    case 'event/CHANGE_EVENT_MOBILE_VIEW':
      return { ...state, isMobileListView: action.isMobileListView }

    default:
      return state
  }
}
