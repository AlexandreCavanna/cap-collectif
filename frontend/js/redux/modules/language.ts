import type { Exact, Action } from '~/types'
import config from '../../config'
export type State = {
  readonly currentLanguage: string
}
const initialState: State = {
  currentLanguage: config.canUseDOM ? new URLSearchParams(window.location.search).get('tl') || window.locale : 'fr_FR',
}
type ChangeLanguageSelectedAction = {
  type: 'language/CHANGE_LANGUAGE_SELECTED'
  currentLanguage: string
}
export type LanguageAction = ChangeLanguageSelectedAction
export const changeEventSelected = (currentLanguage: string): LanguageAction => ({
  type: 'language/CHANGE_LANGUAGE_SELECTED',
  currentLanguage,
})
export const reducer = (state: State = initialState, action: Action): Exact<State> => {
  switch (action.type) {
    case '@@INIT':
      return { ...initialState, ...state }

    case 'language/CHANGE_LANGUAGE_SELECTED':
      return { ...state, currentLanguage: action.currentLanguage }

    default:
      return state
  }
}
