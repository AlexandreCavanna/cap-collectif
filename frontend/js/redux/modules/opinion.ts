import type { Exact, Uuid, Action } from '../../types'
type ShowSourceCreateModalAction = {
  type: 'opinion/SHOW_SOURCE_CREATE_MODAL'
}
type HideSourceCreateModalAction = {
  type: 'opinion/HIDE_SOURCE_CREATE_MODAL'
}
type HideArgumentEditModalAction = {
  type: 'opinion/HIDE_ARGUMENT_EDIT_MODAL'
}
type ShowArgumentEditModalAction = {
  type: 'opinion/SHOW_ARGUMENT_EDIT_MODAL'
  id: Uuid
}
type ShowSourceEditModalAction = {
  type: 'opinion/SHOW_SOURCE_EDIT_MODAL'
  id: Uuid
}
type HideSourceEditModalAction = {
  type: 'opinion/HIDE_SOURCE_EDIT_MODAL'
}
type StartEditOpinionVersionAction = {
  type: 'opinion/START_EDIT_OPINION_VERSION'
}
type CancelEditOpinionVersionAction = {
  type: 'opinion/CANCEL_EDIT_OPINION_VERSION'
}
type StartCreateOpinionVersionAction = {
  type: 'opinion/START_CREATE_OPINION_VERSION'
}
type CancelCreateOpinionVersionAction = {
  type: 'opinion/CANCEL_CREATE_OPINION_VERSION'
}
type ShowOpinionVersionEditModalAction = {
  type: 'opinion/SHOW_OPINION_VERSION_EDIT_MODAL'
}
type CloseOpinionVersionEditModalAction = {
  type: 'opinion/CLOSE_OPINION_VERSION_EDIT_MODAL'
}
type ShowOpinionVersionCreateModalAction = {
  type: 'opinion/SHOW_OPINION_VERSION_CREATE_MODAL'
}
type CloseOpinionVersionCreateModalAction = {
  type: 'opinion/CLOSE_OPINION_VERSION_CREATE_MODAL'
}
type ShowOpinionCreateModalAction = {
  type: 'opinion/SHOW_OPINION_CREATE_MODAL'
  opinionTypeId: Uuid
}
type CloseOpinionCreateModalAction = {
  type: 'opinion/CLOSE_OPINION_CREATE_MODAL'
}
type ShowOpinionEditModalAction = {
  type: 'opinion/SHOW_OPINION_EDIT_MODAL'
  opinionId: Uuid
}
type CloseOpinionEditModalAction = {
  type: 'opinion/CLOSE_OPINION_EDIT_MODAL'
}
export type OpinionAction =
  | ShowSourceCreateModalAction
  | HideSourceCreateModalAction
  | HideSourceEditModalAction
  | HideArgumentEditModalAction
  | ShowArgumentEditModalAction
  | StartEditOpinionVersionAction
  | CancelEditOpinionVersionAction
  | ShowOpinionVersionEditModalAction
  | CloseOpinionVersionEditModalAction
  | ShowOpinionVersionCreateModalAction
  | CloseOpinionVersionCreateModalAction
  | StartCreateOpinionVersionAction
  | CancelCreateOpinionVersionAction
  | ShowOpinionCreateModalAction
  | CloseOpinionCreateModalAction
  | ShowOpinionEditModalAction
  | ShowSourceEditModalAction
  | CloseOpinionEditModalAction
export type State = {
  readonly showOpinionCreateModal: Uuid | null | undefined
  readonly showOpinionEditModal: Uuid | null | undefined
  readonly showArgumentEditModal: Uuid | null | undefined
  readonly showOpinionVersionEditModal: boolean
  readonly isCreatingOpinionVersion: boolean
  readonly showOpinionVersionCreateModal: boolean
  readonly showSourceCreateModal: boolean
  readonly showSourceEditModal: Uuid | null | undefined
}
export const OPINION_VOTE_SUCCEEDED = 'opinion/OPINION_VOTE_SUCCEEDED'
export const VERSION_VOTE_SUCCEEDED = 'opinion/VERSION_VOTE_SUCCEEDED'
export const DELETE_OPINION_VOTE_SUCCEEDED = 'opinion/DELETE_OPINION_VOTE_SUCCEEDED'
export const DELETE_VERSION_VOTE_SUCCEEDED = 'opinion/DELETE_VERSION_VOTE_SUCCEEDED'
const initialState: State = {
  showArgumentEditModal: null,
  showOpinionCreateModal: null,
  showOpinionEditModal: null,
  showOpinionVersionEditModal: false,
  isCreatingOpinionVersion: false,
  showOpinionVersionCreateModal: false,
  showSourceCreateModal: false,
  showSourceEditModal: null,
}
export const openArgumentEditModal = (id: Uuid): ShowArgumentEditModalAction => ({
  type: 'opinion/SHOW_ARGUMENT_EDIT_MODAL',
  id,
})
export const showSourceEditModal = (id: Uuid): ShowSourceEditModalAction => ({
  type: 'opinion/SHOW_SOURCE_EDIT_MODAL',
  id,
})
export const hideSourceEditModal = (): HideSourceEditModalAction => ({
  type: 'opinion/HIDE_SOURCE_EDIT_MODAL',
})
export const showSourceCreateModal = (): ShowSourceCreateModalAction => ({
  type: 'opinion/SHOW_SOURCE_CREATE_MODAL',
})
export const hideSourceCreateModal = (): HideSourceCreateModalAction => ({
  type: 'opinion/HIDE_SOURCE_CREATE_MODAL',
})
export const closeArgumentEditModal = (): HideArgumentEditModalAction => ({
  type: 'opinion/HIDE_ARGUMENT_EDIT_MODAL',
})
export const startCreatingOpinionVersion = (): StartCreateOpinionVersionAction => ({
  type: 'opinion/START_CREATE_OPINION_VERSION',
})
export const cancelCreatingOpinionVersion = (): CancelCreateOpinionVersionAction => ({
  type: 'opinion/CANCEL_CREATE_OPINION_VERSION',
})
export const showOpinionVersionEditModal = (): ShowOpinionVersionEditModalAction => ({
  type: 'opinion/SHOW_OPINION_VERSION_EDIT_MODAL',
})
export const closeOpinionVersionEditModal = (): CloseOpinionVersionEditModalAction => ({
  type: 'opinion/CLOSE_OPINION_VERSION_EDIT_MODAL',
})
export const closeOpinionVersionCreateModal = (): CloseOpinionVersionCreateModalAction => ({
  type: 'opinion/CLOSE_OPINION_VERSION_CREATE_MODAL',
})
export const showOpinionVersionCreateModal = (): ShowOpinionVersionCreateModalAction => ({
  type: 'opinion/SHOW_OPINION_VERSION_CREATE_MODAL',
})
export const openOpinionCreateModal = (opinionTypeId: Uuid): ShowOpinionCreateModalAction => ({
  type: 'opinion/SHOW_OPINION_CREATE_MODAL',
  opinionTypeId,
})
export const closeOpinionCreateModal = (): CloseOpinionCreateModalAction => ({
  type: 'opinion/CLOSE_OPINION_CREATE_MODAL',
})
export const openOpinionEditModal = (opinionId: Uuid): ShowOpinionEditModalAction => ({
  type: 'opinion/SHOW_OPINION_EDIT_MODAL',
  opinionId,
})
export const closeOpinionEditModal = (): CloseOpinionEditModalAction => ({
  type: 'opinion/CLOSE_OPINION_EDIT_MODAL',
})
export const reducer = (state: State = initialState, action: Action): Exact<State> => {
  switch (action.type) {
    case '@@INIT':
      return { ...initialState, ...state }

    case 'opinion/SHOW_SOURCE_EDIT_MODAL': {
      return { ...state, showSourceEditModal: action.id }
    }

    case 'opinion/HIDE_SOURCE_EDIT_MODAL': {
      return { ...state, showSourceEditModal: null }
    }

    case 'opinion/SHOW_SOURCE_CREATE_MODAL': {
      return { ...state, showSourceCreateModal: true }
    }

    case 'opinion/HIDE_SOURCE_CREATE_MODAL': {
      return { ...state, showSourceCreateModal: false }
    }

    case 'opinion/SHOW_ARGUMENT_EDIT_MODAL': {
      return { ...state, showArgumentEditModal: action.id }
    }

    case 'opinion/HIDE_ARGUMENT_EDIT_MODAL': {
      return { ...state, showArgumentEditModal: null }
    }

    case 'opinion/START_CREATE_OPINION_VERSION': {
      return { ...state, isCreatingOpinionVersion: true }
    }

    case 'opinion/CANCEL_CREATE_OPINION_VERSION': {
      return { ...state, isCreatingOpinionVersion: false }
    }

    case 'opinion/SHOW_OPINION_VERSION_EDIT_MODAL': {
      return { ...state, showOpinionVersionEditModal: true }
    }

    case 'opinion/CLOSE_OPINION_VERSION_EDIT_MODAL': {
      return { ...state, showOpinionVersionEditModal: false }
    }

    case 'opinion/SHOW_OPINION_VERSION_CREATE_MODAL': {
      return { ...state, showOpinionVersionCreateModal: true }
    }

    case 'opinion/CLOSE_OPINION_EDIT_MODAL':
      return { ...state, showOpinionEditModal: null }

    case 'opinion/SHOW_OPINION_EDIT_MODAL':
      return { ...state, showOpinionEditModal: action.opinionId }

    case 'opinion/CLOSE_OPINION_CREATE_MODAL':
      return { ...state, showOpinionCreateModal: null }

    case 'opinion/SHOW_OPINION_CREATE_MODAL':
      return { ...state, showOpinionCreateModal: action.opinionTypeId }

    case 'opinion/CLOSE_OPINION_VERSION_CREATE_MODAL': {
      return { ...state, showOpinionVersionCreateModal: false }
    }

    default:
      return state
  }
}
