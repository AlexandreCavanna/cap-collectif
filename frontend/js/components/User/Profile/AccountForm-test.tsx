/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { AccountForm } from './AccountForm'
import { features } from '~/redux/modules/default'
import { intlMock, formMock, $refType, $fragmentRefs } from '~/mocks'

describe('<AccountForm />', () => {
  const defaultViewer = {
    ' $refType': $refType,
    ' $fragmentRefs': $fragmentRefs,
    locale: 'fr-FR',
    email: 'initial-email@gmail.fr',
    facebookId: null,
    isFranceConnectAccount: false,
    newEmailToConfirm: null,
    hasPassword: true,
  }
  const props = {
    ...formMock,
    currentLanguage: 'fr-FR',
    features,
    ssoList: [],
    languageList: [
      {
        translationKey: 'french',
        code: 'fr-FR',
      },
      {
        translationKey: 'english',
        code: 'en-GB',
      },
    ],
    viewer: defaultViewer,
    loginWithOpenId: false,
    intl: intlMock,
    dispatch: jest.fn(),
    handleSubmit: jest.fn(),
    initialValues: {
      email: 'initial-email@gmail.fr',
    },
  }
  it('should render a form', () => {
    const wrapper = shallow(<AccountForm {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render a form with an alert if submitted password is wrong', () => {
    const wrapper = shallow(<AccountForm {...props} error="user.confirm.wrong_password" />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render a form with associated account', () => {
    const franceConnectViewer = { ...defaultViewer, hasPassword: false, isFranceConnectAccount: true }
    const Props = { ...props, viewer: { ...franceConnectViewer } }
    const wrapper = shallow(<AccountForm {...Props} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render a form with information about new email', () => {
    const wrapper = shallow(<AccountForm {...props} newEmailToConfirm="new-email@test.com" />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render a form with an info if a confirmation email has been resent', () => {
    const wrapper = shallow(<AccountForm {...props} newEmailToConfirm="new-email@test.com" />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render with button to delete account only', () => {
    const wrapper = shallow(<AccountForm {...props} loginWithOpenId newEmailToConfirm="new-email@test.com" />)
    expect(wrapper).toMatchSnapshot()
  })
})
