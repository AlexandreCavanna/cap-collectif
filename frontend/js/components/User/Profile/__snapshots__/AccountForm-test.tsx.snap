// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountForm /> should render a form 1`] = `
<Fragment>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    <form
      id="profile-account"
      onSubmit={[MockFunction]}
    >
      <AccountForm__AccountContainer>
        <label
          className="col-sm-3 control-label"
          htmlFor="account__email"
        >
          <FormattedMessage
            id="global.email"
            values={Object {}}
          />
        </label>
        <Field
          component={[Function]}
          disabled={false}
          divClassName="col-sm-6"
          id="account__email"
          name="email"
          type="email"
        />
        <span
          className="account-form-hint"
        >
          <FormattedMessage
            id="account.your_email_is_not_public"
            values={Object {}}
          />
        </span>
        <Connect(ConfirmPasswordModal)
          handleClose={[Function]}
          show={false}
        />
        <div
          className="col-sm-6 col-sm-offset-3 mt-5 mb-15 w-100"
          id="profile-alert-form"
        >
          <AlertForm
            errorMessage={null}
            invalid={false}
            submitFailed={false}
            submitSucceeded={false}
            submitting={false}
            valid={true}
          />
        </div>
      </AccountForm__AccountContainer>
    </form>
  </PanelBody>
  <PanelFooter
    bsClass="panel"
  >
    <div
      className="pl-15"
    >
      <AccountForm__FooterContainer>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          id="edit-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="global.save_modifications"
            values={Object {}}
          />
        </Button>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="danger"
          className="ml-15"
          disabled={false}
          id="delete-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="delete-account"
            values={Object {}}
          />
        </Button>
      </AccountForm__FooterContainer>
      <Relay(DeleteAccountModal)
        handleClose={[Function]}
        show={false}
        viewer={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
            "email": "initial-email@gmail.fr",
            "facebookId": null,
            "hasPassword": true,
            "isFranceConnectAccount": false,
            "locale": "fr-FR",
            "newEmailToConfirm": null,
          }
        }
      />
    </div>
  </PanelFooter>
</Fragment>
`;

exports[`<AccountForm /> should render a form with an alert if submitted password is wrong 1`] = `
<Fragment>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    <form
      id="profile-account"
      onSubmit={[MockFunction]}
    >
      <AccountForm__AccountContainer>
        <label
          className="col-sm-3 control-label"
          htmlFor="account__email"
        >
          <FormattedMessage
            id="global.email"
            values={Object {}}
          />
        </label>
        <Field
          component={[Function]}
          disabled={false}
          divClassName="col-sm-6"
          id="account__email"
          name="email"
          type="email"
        />
        <span
          className="account-form-hint"
        >
          <FormattedMessage
            id="account.your_email_is_not_public"
            values={Object {}}
          />
        </span>
        <Connect(ConfirmPasswordModal)
          handleClose={[Function]}
          show={false}
        />
        <div
          className="col-sm-6 col-sm-offset-3 mt-5 mb-15 w-100"
          id="profile-alert-form"
        >
          <AlertForm
            errorMessage="user.confirm.wrong_password"
            invalid={false}
            submitFailed={false}
            submitSucceeded={false}
            submitting={false}
            valid={true}
          />
        </div>
      </AccountForm__AccountContainer>
    </form>
  </PanelBody>
  <PanelFooter
    bsClass="panel"
  >
    <div
      className="pl-15"
    >
      <AccountForm__FooterContainer>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          id="edit-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="global.save_modifications"
            values={Object {}}
          />
        </Button>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="danger"
          className="ml-15"
          disabled={false}
          id="delete-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="delete-account"
            values={Object {}}
          />
        </Button>
      </AccountForm__FooterContainer>
      <Relay(DeleteAccountModal)
        handleClose={[Function]}
        show={false}
        viewer={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
            "email": "initial-email@gmail.fr",
            "facebookId": null,
            "hasPassword": true,
            "isFranceConnectAccount": false,
            "locale": "fr-FR",
            "newEmailToConfirm": null,
          }
        }
      />
    </div>
  </PanelFooter>
</Fragment>
`;

exports[`<AccountForm /> should render a form with an info if a confirmation email has been resent 1`] = `
<Fragment>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    <form
      id="profile-account"
      onSubmit={[MockFunction]}
    >
      <AccountForm__AccountContainer>
        <label
          className="col-sm-3 control-label"
          htmlFor="account__email"
        >
          <FormattedMessage
            id="global.email"
            values={Object {}}
          />
        </label>
        <Field
          component={[Function]}
          disabled={false}
          divClassName="col-sm-6"
          id="account__email"
          name="email"
          type="email"
        />
        <span
          className="account-form-hint"
        >
          <FormattedMessage
            id="account.your_email_is_not_public"
            values={Object {}}
          />
        </span>
        <Connect(ConfirmPasswordModal)
          handleClose={[Function]}
          show={false}
        />
        <div
          className="col-sm-6 col-sm-offset-3 mt-5 mb-15 w-100"
          id="profile-alert-form"
        >
          <AlertForm
            errorMessage={null}
            invalid={false}
            submitFailed={false}
            submitSucceeded={false}
            submitting={false}
            valid={true}
          />
        </div>
      </AccountForm__AccountContainer>
    </form>
  </PanelBody>
  <PanelFooter
    bsClass="panel"
  >
    <div
      className="pl-15"
    >
      <AccountForm__FooterContainer>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          id="edit-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="global.save_modifications"
            values={Object {}}
          />
        </Button>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="danger"
          className="ml-15"
          disabled={false}
          id="delete-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="delete-account"
            values={Object {}}
          />
        </Button>
      </AccountForm__FooterContainer>
      <Relay(DeleteAccountModal)
        handleClose={[Function]}
        show={false}
        viewer={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
            "email": "initial-email@gmail.fr",
            "facebookId": null,
            "hasPassword": true,
            "isFranceConnectAccount": false,
            "locale": "fr-FR",
            "newEmailToConfirm": null,
          }
        }
      />
    </div>
  </PanelFooter>
</Fragment>
`;

exports[`<AccountForm /> should render a form with associated account 1`] = `
<Fragment>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    <form
      id="profile-account"
      onSubmit={[MockFunction]}
    >
      <AccountForm__AccountContainer>
        <label
          className="col-sm-3 control-label"
          htmlFor="account__email"
        >
          <FormattedMessage
            id="global.email"
            values={Object {}}
          />
        </label>
        <Field
          component={[Function]}
          disabled={false}
          divClassName="col-sm-6"
          id="account__email"
          name="email"
          type="email"
        />
        <span
          className="account-form-hint"
        >
          <FormattedMessage
            id="data-from-FranceConnect"
            values={Object {}}
          />
          <FormattedMessage
            id="account.your_email_is_not_public"
            values={Object {}}
          />
        </span>
        <Connect(ConfirmPasswordModal)
          handleClose={[Function]}
          show={false}
        />
        <div
          className="col-sm-6 col-sm-offset-3 mt-5 mb-15 w-100"
          id="profile-alert-form"
        >
          <AlertForm
            errorMessage={null}
            invalid={false}
            submitFailed={false}
            submitSucceeded={false}
            submitting={false}
            valid={true}
          />
        </div>
      </AccountForm__AccountContainer>
    </form>
  </PanelBody>
  <PanelFooter
    bsClass="panel"
  >
    <div
      className="pl-15"
    >
      <AccountForm__FooterContainer>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          id="edit-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="global.save_modifications"
            values={Object {}}
          />
        </Button>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="danger"
          className="ml-15"
          disabled={false}
          id="delete-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="delete-account"
            values={Object {}}
          />
        </Button>
      </AccountForm__FooterContainer>
      <Relay(DeleteAccountModal)
        handleClose={[Function]}
        show={false}
        viewer={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
            "email": "initial-email@gmail.fr",
            "facebookId": null,
            "hasPassword": false,
            "isFranceConnectAccount": true,
            "locale": "fr-FR",
            "newEmailToConfirm": null,
          }
        }
      />
    </div>
  </PanelFooter>
</Fragment>
`;

exports[`<AccountForm /> should render a form with information about new email 1`] = `
<Fragment>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    <form
      id="profile-account"
      onSubmit={[MockFunction]}
    >
      <AccountForm__AccountContainer>
        <label
          className="col-sm-3 control-label"
          htmlFor="account__email"
        >
          <FormattedMessage
            id="global.email"
            values={Object {}}
          />
        </label>
        <Field
          component={[Function]}
          disabled={false}
          divClassName="col-sm-6"
          id="account__email"
          name="email"
          type="email"
        />
        <span
          className="account-form-hint"
        >
          <FormattedMessage
            id="account.your_email_is_not_public"
            values={Object {}}
          />
        </span>
        <Connect(ConfirmPasswordModal)
          handleClose={[Function]}
          show={false}
        />
        <div
          className="col-sm-6 col-sm-offset-3 mt-5 mb-15 w-100"
          id="profile-alert-form"
        >
          <AlertForm
            errorMessage={null}
            invalid={false}
            submitFailed={false}
            submitSucceeded={false}
            submitting={false}
            valid={true}
          />
        </div>
      </AccountForm__AccountContainer>
    </form>
  </PanelBody>
  <PanelFooter
    bsClass="panel"
  >
    <div
      className="pl-15"
    >
      <AccountForm__FooterContainer>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          id="edit-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="global.save_modifications"
            values={Object {}}
          />
        </Button>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="danger"
          className="ml-15"
          disabled={false}
          id="delete-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="delete-account"
            values={Object {}}
          />
        </Button>
      </AccountForm__FooterContainer>
      <Relay(DeleteAccountModal)
        handleClose={[Function]}
        show={false}
        viewer={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
            "email": "initial-email@gmail.fr",
            "facebookId": null,
            "hasPassword": true,
            "isFranceConnectAccount": false,
            "locale": "fr-FR",
            "newEmailToConfirm": null,
          }
        }
      />
    </div>
  </PanelFooter>
</Fragment>
`;

exports[`<AccountForm /> should render with button to delete account only 1`] = `
<Fragment>
  <PanelFooter
    bsClass="panel"
  >
    <div
      className="pl-15"
    >
      <AccountForm__FooterContainer>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="danger"
          className="ml-15"
          disabled={false}
          id="delete-account-profile-button"
          onClick={[Function]}
        >
          <FormattedMessage
            id="delete-account"
            values={Object {}}
          />
        </Button>
      </AccountForm__FooterContainer>
      <Relay(DeleteAccountModal)
        handleClose={[Function]}
        show={false}
        viewer={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
            "email": "initial-email@gmail.fr",
            "facebookId": null,
            "hasPassword": true,
            "isFranceConnectAccount": false,
            "locale": "fr-FR",
            "newEmailToConfirm": null,
          }
        }
      />
    </div>
  </PanelFooter>
</Fragment>
`;
